### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by Hoxbro**
This PR removes old deprecations:

- Non-working tile sources
- Having a non-string as a column name in a DataFrame

I also added warnings.py (and removed _exceptions.py) to align how Panel does it.

### CODE REVIEW

File: holoviews/element/tiles.py [hunk 1]
```diff
@@ -127,27 +126,10 @@ def easting_northing_to_lon_lat(easting, northing):
     )
 }
 
-def deprecation_warning(name, url, reason):
-    def deprecated_tilesource_warning():
```
Comment (jlstevens) [inline holoviews/element/tiles.py:131]: This deprecation approach for tile sources was added not so long ago...removing it now might be a bit too soon!

**Comment by jlstevens** [comment]
Made one quick comment, otherwise looks good!

File: holoviews/element/tiles.py (hunk 1 above)
Comment (Hoxbro) [reply holoviews/element/tiles.py:131]: This was added two years ago is that too soon?

https://github.com/holoviz/holoviews/pull/4941

Comment (Hoxbro) [reply holoviews/element/tiles.py:131]: The notebook is working. Though, I update the text to reflect it.

File: examples/reference/elements/bokeh/Tiles.ipynb [hunk 2]
```diff
@@ -77,7 +77,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The full set of predefined tile sources can be accessed on the ``holoviews.element.tiles.tile_sources`` dictionary. Some tile sources from CartoDB are no longer publicly available and marked with a deprecation warning. These are filtered below:"
+    "The full set of predefined tile sources can be accessed on the `holoviews.element.tiles.tile_sources` dictionary. "
```
Comment (Hoxbro) [inline examples/reference/elements/bokeh/Tiles.ipynb:80]:
```suggestion
    "The full set of predefined tile sources can be accessed on the ``holoviews.element.tiles.tile_sources`` dictionary."
```

File: holoviews/element/tiles.py (hunk 1 above)
Comment (jlstevens) [reply holoviews/element/tiles.py:131]: Thanks!

**Pull request merged**